### DISCUSSION TITLE
How do you put Images on the README.md file?

### DISCUSSION THREAD

**Discussion by SHADOWELITE7** [community/community#22833] [New to GitHub]
<p>I was asking because I see a lot of repositories and tried doing the same thing but couldn’t figure out how to do it</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/shadowelite7">@shadowelite7</a>,</p>
<p>README.md files are created using Markdown which you can use to format text and add images. <a href="https://guides.github.com/features/mastering-markdown/" rel="nofollow noopener">The GitHub Guide for Mastering Markdown</a> has some helpful hints that can get you started with this.</p>
<p>Thanks!</p>

**Comment by mpboom** [comment]
<p><img alt="Image description"></p>

**Comment by SHADOWELITE7** [comment]
<p>thx</p>
<blockquote>
<p>wrote:</p>
</blockquote>
<blockquote>
<p><img alt="Image description"></p>
</blockquote>

**Comment by SHADOWELITE7** [comment]
<p>thx</p>
<blockquote>
<p>wrote:</p>
</blockquote>
<blockquote>
<p>Hi <a class="mention" href="/shadowelite7">@shadowelite7</a>,</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>README.md files are created using Markdown which you can use to format text and add images. <a href="https://guides.github.com/features/mastering-markdown/" rel="nofollow noopener">The GitHub Guide for Mastering Markdown</a> has some helpful hints that can get you started with this.</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>Thanks!</p>
</blockquote>

**Comment by tograh** [comment]
<p>Does this still work?  Because I’m trying the same thing, then navigating to the github page, and all I see are broken image icons, even though some display fine locally.  I expected “Test Image 1” to work when the file 3DTest.png is uploaded into the same directory.</p>
<p>![Test Image 1](3DTest.png)</p>
<p>![Test Image 2](“3DTest.png”)</p>
<p>![Test Image 3](/3DTest.png)</p>
<p>![Test Image 4](<a href="https://github.com/tograh/testrepository/3DTest.png" target="_blank" rel="nofollow noopener">https://github.com/tograh/testrepository/3DTest.png</a>)</p>
<p>![Test Image 5](https://…/3DTest.png)</p>
<p>![Test Image 6](master/3DTest.png)</p>
<p>![Test Image 7](<a href="https://github.com/tograh/testrepository/master/3DTest.png" target="_blank" rel="nofollow noopener">https://github.com/tograh/testrepository/master/3DTest.png</a>)</p>
<p>![Test Image 8](<a href="https://raw.githubusercontent.com/tograh/testrepository/master/3DTest.png" target="_blank" rel="nofollow noopener">https://raw.githubusercontent.com/tograh/testrepository/master/3DTest.png</a>)</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/tograh">@tograh</a>,</p>
<p>For the images to work properly, you have to include the full image URL. That said, if none of your images are working, I’d be curious to see the repo in which they are located, if possible. Are you doing this work in a public repo? If so, can you provide a link so I/we can take a closer look?</p>

**Reply by Atlas-Night-Out** [reply]
Be great if you can help me do images in github to then a link text to it to just show the image and not it in the repo! Then I also need resize fo rit too, tried lots of ways and examples and no luck yet on what I need! If you can help will be great

**Reply by mpboom** [reply]
@Atlas-Night-Out instead of commenting on a 7-year old thread, consider opening a new discussion.

**Comment by tograh** [comment]
<p>Thank you for the reply, Nadia.  We’ve been looking into the problem, and it seems to be limited to inside our corporate network.  Outside, we see the markup displaying as expected.</p>
<p>I should have updated sooner!  Many thanks.</p>

**Comment by KrisKasprzak** [comment]
<p>OK i give up. how do you know the full path?</p>
<p>tried both of these. my images are in the top level folder</p>
<p>&lt;img src=“<a href="https://github.com/KrisKasprzak/ILI9341_t3_controls" rel="nofollow noopener">https://github.com/KrisKasprzak/ILI9341_t3_controls</a>”&gt;<br>
<img alt="Equalizer"></p>

**Reply by Tanntech** [reply]
same icon is coming even i tried 10 times

**Comment by mpboom** [comment]
<p>Hi <a class="mention" href="/kriskasprzak">@kriskasprzak</a> ,</p>
<p>In your case it would be:</p>
<p><a href="https://raw.githubusercontent.com/KrisKasprzak/ILI9341_t3_controls/master/Equalizer.BMP" rel="nofollow noopener">https://raw.githubusercontent.com/KrisKasprzak/ILI9341_t3_controls/master/Equalizer.BMP</a></p>
<p>You can obtain the link by going to:</p>
<p><a href="https://github.com/KrisKasprzak/ILI9341_t3_controls/blob/master/Equalizer.BMP" rel="nofollow noopener">https://github.com/KrisKasprzak/ILI9341_t3_controls/blob/master/Equalizer.BMP</a></p>
<p>And then clicking  <strong>View raw</strong>.</p>

**Comment by KrisKasprzak** [comment]
<p>that did not work.</p>
<p>apparently the site will not use 16-bit BMP’s I had to convert to jpg</p>
<p>this actually worked</p>
<p><img alt="header image"><br>
<img alt="header image"></p>

**Comment by mpboom** [comment]
<p>Oh, I didn’t know that. Thanks for sharing your findings with the community!</p>

**Comment by antt179** [comment]
<p>Need to buy Github account which register before 1/9/2019</p>

**Comment by SwingGuy1024** [comment]
<p>Here’s the thing. I want to get my link right before I check the file into my github repository. But it looks like I cann’t make the link work until after I check it in! Is there any way to test my link before I check anything in? </p>
<p>Right now, the graphic is not check in, and the .md file is there, but without the new link.</p>

**Comment by okontsograda** [comment]
<p>This did the trick nicely! Upload the file to the same directory and just use this to inludie it within the readme. Thanks a lot, this saved me a lot of searching around for a solution.</p>

**Comment by sdg002** [comment]
<p>For local images I used this:</p>
<p><img alt=""></p>
<p>I am torn. I liked the initial simplicity of Markdown. However, I have now begun to question the merits of developing a radically new language/syntax which appears to rapidly solve some low hanging problems but leaves a trail of undocumented or poor documented issues as we go delve deeper. A part of me wonders whether we should go back to HTML -  whos syntax is rock solid and the ecosystem of editors very well developed.</p>

**Comment by tiagotakeda** [comment]
<p>I got a similar issue and when I added:</p>
<p>raw=true</p>
<p>it worked. So the code, in HTML, for my imagem was:</p>
<p>&lt;img<br>
src=“images/SubPronouns.jpg”<br>
raw=true<br>
alt=“Subject Pronouns”<br>
style=“margin-right: 10px;”<br>
/&gt;</p>

**Comment by ruminize** [comment]
<p>With the image in the git repo root, this worked for me.</p>
<p>&lt;img src="./myimage.jpg"&gt;</p>

**Comment by huzaifsayed** [comment]
<p>Using This Command </p>
<p><img alt="Image description"></p>

**Reply by Tanntech** [reply]
<img src="https://github.com/Tanntech/Tanntech/blob/main/GitHubBannernew.png">

**Comment by jaimehernan95** [comment]
<p>I tried the solutions given below, but I didn’t work for me. I uploaded the image from the command line.</p>
<p>1 I created a folder with images </p>
<ol start="2">
<li>
<p>Navigate to the image folder and right-click to get the image link</p>
</li>
<li>
<p>go to your README file and paste the link </p>
</li>
<li>
<p>By doing this GitHub will generate automatically the perfect width to fit your page.</p>
</li>
</ol>
<p>Example</p>
<p>Images src link:</p>
<p>&lt;img width=“964” alt=“java 8 and prio java 8  array review example” src=“<a href="https://github.com/jaimehernan95/arrayReview-java/blob/master/images/%20arrayExample.png" rel="nofollow noopener">https://github.com/jaimehernan95/arrayReview-java/blob/master/images/%20arrayExample.png</a>”&gt;</p>
<p>README. File</p>
<p><a href="https://github.com/jaimehernan95/arrayReview-java/blob/master/README.md" rel="nofollow noopener">https://github.com/jaimehernan95/arrayReview-java/blob/master/README.md</a></p>
<p>jaime</p>

**Comment by Teuton-Roy** [comment]
It's very simple using [ctrl+c / ctrl+v]

**Comment by anuproy** [comment]
What is the solution if I upload an image with the caption?

**Reply by mpboom** [reply]
After pasting the image in, you can just add the caption as usual in the editor.

**Reply by anuproy** [reply]
Thanks for your reply @mpboom 
` <figcaption>Fig.1 </figcaption>`
I hope this can be applicable.

**Comment by tracymarshallaz** [comment]
#![image](https://user-images.githubusercontent.com/92385352/216928213-324bf903-e358-4c8a-b679-f1d695f6f8d1.jpeg)

**Reply by anuproy** [reply]
What is this? maybe you have shared the wrong place @tracymarshallaz

**Comment by bilalkhan5001** [comment]
To add images to a `README.md` file in your GitHub repository, follow these steps:

---

**1. Use Markdown Syntax**
The basic syntax for embedding an image in Markdown is:  
```markdown
![Alt text](image-url)
```

- **`Alt text`**: A description of the image (displayed if the image can't load).  
- **`image-url`**: The URL or relative path to the image.

---

### **2. Host the Image**
You can use an image from the following sources:  

#### **A. Uploaded to Your Repository**  
1. Add the image file (e.g., `image.png`) to your repository.  
2. Use its relative path:  
   ```markdown
   ![Alt text](images/image.png)
   ```
   Replace `images/image.png` with the actual file path.

#### **B. External URL**  
Use an image hosted elsewhere by providing the direct link:  
```markdown
![Alt text](https://example.com/image.png)
```

#### **C. GitHub Hosted Image**  
1. Drag and drop the image into a GitHub issue, comment, or discussion.  
2. Copy the generated URL for the uploaded image.  
3. Use it in your `README.md`:  
   ```markdown
   ![Alt text](https://user-images.githubusercontent.com/123456789/your-image.png)
   ```

---

### **3. Advanced Examples**  

#### **With Size Adjustment (HTML)**  
Markdown doesn't support resizing images, but you can use HTML inside your `README.md`:  
```html
<img src="image-url" alt="Alt text" width="300" height="200">
```

#### **Linking an Image**  
You can make an image clickable:  
```markdown
[![Alt text](image-url)](link-url)
```
Example:  
```markdown
[![Logo](images/logo.png)](https://github.com)
```
